### PR TITLE
Use ghost eslint plugin in server side code (#9834)

### LINKED ISSUE #9834: Use ghost eslint plugin
A couple of month ago we have created a [Ghost eslint plugin.](https://github.com/TryGhost/eslint-plugin-ghost).

It's used already in a couple of places, but not on the server side 👻✌🏻

You can take a look at [this commit](https://github.com/TryGhost/Ghost-Admin/commit/7afb88a0c21babe047c91a17c34e01d849079564) and follow the steps to replace the code with the new plugin.

### PR DESCRIPTION
**Pull request by lunaticmonk**
I have followed the same approach as is used [here](https://github.com/TryGhost/Ghost-Admin/commit/7afb88a0c21babe047c91a17c34e01d849079564). However, some rules had to be overridden in order to pass the eslint on server side.

Fixes #9834 

- [x] There's a clear use-case for this code change
- [x] Commit message has a short title & references relevant issues
- [ ] The build will pass (run `npm test`)

More info can be found by clicking the "guidelines for contributing" link above.

### CODE REVIEW

**Comment by ErisDS** [comment]
Hey @lunaticmonk, thanks for picking this up 👍 

`eslint-plugin-ghost` has multiple configs. We use `ember` in the client because that is an ember app, but it would be correct to use `node` here, i.e. `"plugin:ghost/node"` instead of `"plugin:ghost/ember"` then you shouldn't have to override rules.

**Comment by lunaticmonk** [comment]
@ErisDS My bad. I forgot to change the interpreter there. I have made the changes. Please review.

**Comment by AileenCGN** [comment]
Hey @lunaticmonk 👋 

The linting task is currently failing, because the plugin is not installed yet. If you could run `yarn add -D eslint-plugin-ghost`, it should work. You can also run then `yarn lint` to check if the code changes pass the linting.

Another thing that's missing is the `.eslintrc.json` file in `/tests/`. This needs to be updated to use the `"plugin:ghost/test"`. Furthermore we need to remove any `jshint` related things (comments, files, etc).

Thanks again for picking this up 🤗. Let me know if you need a hand with that.

**Review by kirrg001 — CHANGES_REQUESTED** [review]
https://github.com/TryGhost/Ghost/pull/9835#issuecomment-419354476 🙂Just updated the status of the PR.

**Comment by lunaticmonk** [comment]
@AileenCGN For the tests' eslintrc.json, I did the changes. However, there are a lot of "no-useless-escape" errors still out there if we don't override any rules. If we declare explicitly in rules to keep the "no-useless-escape" off, this will solve the problem. The thing is, should I correct those all manually now or should we ignore them by overriding the rules for tests?

**Comment by AileenCGN** [comment]
@lunaticmonk I think there are a couple of problems here:

1. The `eslint-plugin-ghost` module is not installed yet
2. The tests are not yet completely refactored to ES6, see https://github.com/TryGhost/Ghost/issues/9589, but the plugin [uses ES6](https://github.com/TryGhost/eslint-plugin-ghost/blob/master/lib/config/base.js#L7)
3. Rather than overwriting rules, we should fix them or at least comment them out for the specific line if applicable

I suggest, that we keep this PR on hold until the refactor to ES6 is completed.

@kirrg001 What do you think?

**Comment by kirrg001** [comment]
I have not used the `eslint-plugin-ghost` plugin on the server side yet. I just noticed that most of the changes from the PR were coming from enforcing this eslint rule: https://eslint.org/docs/rules/arrow-body-style.

I wabt to disable this rule, which should make the ES6 updates smaller. 

Please wait till [this PR](https://github.com/TryGhost/eslint-plugin-ghost/pull/8) got released. Will happen today.

> The tests are not yet completely refactored to ES6, see #9589, but the plugin uses ES6

I'd suggest to first replace the eslint file in the root dir, and another PR can update the test directory.

> I suggest, that we keep this PR on hold until the refactor to ES6 is completed.

Hmm i would love to merge this. IMO you could leave a comment on #9589. @rishabhgrg is currently updating ES6 in the `api` and `web` folder, but it should not conflict much i guess.

**Comment by kirrg001** [comment]
[0.0.27 was released](https://github.com/TryGhost/eslint-plugin-ghost/releases)

**Comment by lunaticmonk** [comment]
@AileenCGN @kirrg001 
I have done the following changes:
- Used `eslint-plugin-ghost` for server side code.
- Fixed the eslint errors.

Please review.
I think the change for the use of `eslint-plugin-ghost` for tests code should be done in other PR.

File: package.json [hunk 1]
```diff
@@ -104,6 +104,7 @@
   },
   "devDependencies": {
     "eslint": "4.19.1",
+    "eslint-plugin-ghost": "^0.0.27",
```
Comment (kirrg001) [inline package.json]: We use exact versions in Ghost.

`yarn add ... --exact ...`

**Review by kirrg001 — APPROVED** [review]
Thanks so much <3

File: yarn.lock [hunk 2]
```diff
@@ -1608,6 +1612,26 @@ escodegen@^1.8.1:
...
     source-map "~0.6.1"
 
+eslint-plugin-ember@^5.0.3:
+  version "5.2.0"
+  resolved "https://registry.yarnpkg.com/eslint-plugin-ember/-/eslint-plugin-ember-5.2.0.tgz#fa436e0497dfc01d1d38608229cd616e7c5b6067"
+  dependencies:
+    ember-rfc176-data "^0.3.3"
+    snake-case "^2.1.0"
+
+eslint-plugin-ghost@^0.0.27:
```
Comment (kirrg001) [inline yarn.lock]: Whoops just saw this. 

You need to re-generate the file or re-add the dep correctly, because you initially added `^0.0.27`.

Comment (lunaticmonk) [reply yarn.lock]: Ohh wait. will check.

Comment (lunaticmonk) [reply yarn.lock]: @kirrg001 done.

**Pull request merged**